File: src/dotnet/commands/dotnet-new/VisualBasic_Web/Controllers/HomeController.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Threading.Tasks
Imports Microsoft.AspNetCore.Mvc

Namespace MvcApp.Controllers
    Public Class HomeController
        Inherits Controller

        Public Function Index() As IActionResult
            Return View()
        End Function

        Public Function About() As IActionResult
            ViewData("Message") = "Your application description page."

            Return View()
        End Function

        Public Function Contact() As IActionResult
            ViewData("Message") = "Your contact page."

            Return View()
        End Function

        Public Function [Error]() As IActionResult
            Return View()
        End Function
    End Class
End Namespace

File: src/dotnet/commands/dotnet-new/FSharp_Web/Controllers/HomeController.fs
﻿namespace MvcApp.Controllers

open System
open System.Collections.Generic
open System.Linq
open System.Threading.Tasks
open Microsoft.AspNetCore.Mvc

type HomeController () =
    inherit Controller()

    member this.Index () =
        this.View()

    member this.About () =
        this.ViewData.["Message"] <- "Your application description page."
        this.View()

    member this.Contact () =
        this.ViewData.["Message"] <- "Your contact page."
        this.View()

    member this.Error () =
        this.View();
